Release 4.2.0

Blackbaud-TrevorBurch opened the pull request.

blackbaud-johnly commented on CHANGELOG.md:
```diff
@@ -1,3 +1,9 @@
+# 4.2.0 (2020-07-08)
+
+- Added the ability to disable the autocomplete component via a reactive form. [#101](https://github.com/blackbaud/skyux-lookup/pull/101)
```
Is this only for reactive forms? If so, do we need to update the description in the docs?

Also, should `via` should be `on`? I mean, I assume we're talking about disabling autocomplete components ON reactive forms, not using reactive forms to disable autocomplete components, right?

Blackbaud-AlexKingman commented on CHANGELOG.md:
```diff
@@ -1,3 +1,9 @@
+# 4.2.0 (2020-07-08)
+
+- Added the ability to disable the autocomplete component via a reactive form. [#101](https://github.com/blackbaud/skyux-lookup/pull/101)
```
Should we reword this to point out the new `@Input()` property? Something like ...
```
Added the `disabled` input property to the the `skyAutocomplete` directive.
```

blackbaud-johnly replied:
I like that, but I don't think we need `input`.

blackbaud-johnly replied:
@Blackbaud-TrevorBurch, I'm going to go ahead and make this change. I can always change it back if you it's a problem.

blackbaud-ado commented:
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-release-4.2.0](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-release-4.2.0)

Blackbaud-TrevorBurch commented:
@blackbaud-johnly @Blackbaud-AlexKingman I think both of your concerns were valid. I went back and dug through the changes I made and remembered that "template forms" were also only partially working prior to this change. With that, we added the `@Input` for template forms (though reactive forms could in theory use this as well which is why we don't call out a difference in the docs) along with adding the `setDisabledState` function which enables reactive forms to set the disabled state via the form control and not the component directly. 

Since this change touches both types of forms but in two different ways I have tweaked the changelog to be: 

```
Added the ability to disable the autocomplete component.
```

We can call out the new input and/or function but to me I don't think we necessarily need to when we really added the ability via two different changes that are only somewhat interchangeable. I think adding too many details might just cause confusion when really the root of what we did was add the ability to disable the component.

blackbaud-ado commented:
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-release-4.2.0](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-release-4.2.0)

blackbaud-ado commented:
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-release-4.2.0](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-release-4.2.0)

blackbaud-ado commented:
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-release-4.2.0](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-release-4.2.0)
